dnf/module/repo_module_dict.py

```diff
@@ -0,0 +1,462 @@
+            else:
+                # if version is not specified, pick the latest
+                repo_module_version = repo_module_stream.latest()
+
+            # TODO: arch
+            # TODO: context
+
+        except KeyError:
+            return None
+        return repo_module_version
+
+    def get_includes_latest(self, name, stream):
+        includes = set()
+        try:
+            repo_module = self[name]
+            repo_module_stream = repo_module[stream]
+            repo_module_version = repo_module_stream.latest()
+
+            artifacts = repo_module_version.module_metadata.artifacts.rpms
+            includes.update(artifacts)
+
+            for requires_name, requires_stream in \
+                    repo_module_version.module_metadata.requires.items():
+                includes.update(self.get_includes_latest(requires_name, requires_stream))
+        except KeyError as e:
+            logger.debug(e)
+
+        return includes
+
+    def get_includes(self, name, stream):
+        includes = set()
+        try:
+            repo_module = self[name]
+            repo_module_stream = repo_module[stream]
+            for repo_module_version in repo_module_stream.values():
+                artifacts = repo_module_version.module_metadata.artifacts.rpms
+                includes.update(artifacts)
+
+                for requires_name, requires_stream in \
+                        repo_module_version.module_metadata.requires.items():
+                    includes.update(self.get_includes_latest(requires_name, requires_stream))
+        except KeyError as e:
+            logger.debug(e)
+
+        return includes
+
+    def enable(self, pkg_spec, assumeyes, assumeno=False):
+        subj = ModuleSubject(pkg_spec)
+        module_version, nsvcap = subj.find_module_version(self)
+
+        self[module_version.name].enable(module_version.stream, assumeyes, assumeno)
+
+    def disable(self, pkg_spec):
+        subj = ModuleSubject(pkg_spec)
+        module_version, nsvcap = subj.find_module_version(self)
+
+        repo_module = module_version.repo_module
+        repo_module.disable()
+
+    def lock(self, pkg_spec):
+        subj = ModuleSubject(pkg_spec)
+        module_version, nsvcap = subj.find_module_version(self)
+
+        repo_module = module_version.repo_module
+
+        if not repo_module.conf.enabled:
+            raise Error(module_errors[STREAM_NOT_ENABLED_ERR].format(pkg_spec))
+
+        repo_module.lock(module_version.version)
+        return repo_module.conf.stream, repo_module.conf.version
+
+    def unlock(self, pkg_spec):
+        subj = ModuleSubject(pkg_spec)
+        module_version, nsvcap = subj.find_module_version(self)
+
+        repo_module = module_version.repo_module
+
+        if not repo_module.conf.enabled:
+            raise Error(module_errors[STREAM_NOT_ENABLED_ERR].format(pkg_spec))
+
+        repo_module.unlock()
+        return
+
+    def install(self, pkg_specs, assumeyes=False):
```

Comment:
strict option is missing here as parameter, also assumeno should have expected behavior. Here we have a problem, that the function combines two functionality - cli and base api